src/services/sessionExec.js

```diff
@@ -67,6 +67,40 @@ const deleteFeedback = async ({ sessionId, feedbackId, userId }) => {
   return session
 }
 
+/**
+ * Update a feedback from a session
+ * @param {*} param0
+ */
+const updateFeedback = async ({ sessionId, feedbackId, tags, userId }) => {
+  const session = await getRunningSession(sessionId)
+
+  // ensure the user is authorized to modify this session
+  if (!session.user.equals(userId)) {
+    throw new ForbiddenError('UNAUTHORIZED')
+  }
+
+  // add new tag to a feedback
+  const index = session.feedbacks.indexOf(session.feedbacks.find(feedback => feedback.id === feedbackId))
+  session.feedbacks[index].tags = tags
+
+  // sort feedbacks
+  session.feedbacks = session.feedbacks.sort((f1, f2) => {
+    if (f1.tags.length !== 0) {
+      return 1
+    }
+    if (f2.tags.length !== 0) {
+      return -1
+    }
+    return 0
+  })
+
+  // save the updated session
+  await session.save()
```

Comment:
Does saving the updated tags like this actually work? Sometimes if you change deeply nested stuff in MongoDB (like switching out the tags of a feedback in the feedbacks array), you need to tell at what index something changed. Maybe this did not cause issues due to the sorting above but could if it were removed.

Something like `session.markModified("feedbacks.2.tags")` before calling `.save()` to tell mongoose that the tags array for the third feedback has changed.

-> search for `.markModified` in the `src/` folder for examples